src/platform/Ameba/BLEManagerImpl.cpp

```diff
@@ -810,6 +851,169 @@ bool BLEManagerImpl::IsSubscribed(uint16_t conId)
     return false;
 }
 
+void BLEManagerImpl::HandleRXCharWrite(uint8_t * p_value, uint16_t len, uint8_t conn_id)
+{
+    CHIP_ERROR err         = CHIP_NO_ERROR;
+    PacketBufferHandle buf = System::PacketBufferHandle::New(len, 0);
+    memcpy(buf->Start(), p_value, len);
+    buf->SetDataLength(len);
+
+    // Post an event to the Chip queue to deliver the data into the Chip stack.
+    {
+        ChipDeviceEvent event;
+        event.Type                        = DeviceEventType::kCHIPoBLEWriteReceived;
+        event.CHIPoBLEWriteReceived.ConId = (uint16_t) conn_id;
+        event.CHIPoBLEWriteReceived.Data  = std::move(buf).UnsafeRelease();
+        PlatformMgr().PostEventOrDie(&event);
+    }
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "HandleRXCharWrite() failed: %s", ErrorStr(err));
+    }
+}
+
+#if defined(CONFIG_MATTER_BLEMGR_ADAPTER) && CONFIG_MATTER_BLEMGR_ADAPTER
+CHIP_ERROR BLEManagerImpl::matter_blemgr_gap_connect_cb(uint8_t conn_id)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    err            = sInstance.HandleGAPConnect((uint16_t) conn_id);
+    SuccessOrExit(err);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Disabling CHIPoBLE service due to error: %s", ErrorStr(err));
+        sInstance.mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_Disabled;
+    }
+
+    // Schedule DriveBLEState() to run.
+    PlatformMgr().ScheduleWork(DriveBLEState, 0);
+
+    return err;
+}
+
+CHIP_ERROR BLEManagerImpl::matter_blemgr_gap_disconnect_cb(uint8_t conn_id, uint16_t disc_cause)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    err            = sInstance.HandleGAPDisconnect((uint16_t) conn_id, disc_cause);
+    SuccessOrExit(err);
```

Comment:
remove the `SuccessOrExit` and `exit:` label here as well: there seems no need for it.